src/Http/Http.Extensions/src/HttpResponseJsonExtensions.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+#nullable enable
+
+namespace Microsoft.AspNetCore.Http.Json
+{
+    public static partial class HttpResponseJsonExtensions
+    {
+        public static Task WriteAsJsonAsync<TValue>(
+            this HttpResponse response,
+            [AllowNull] TValue value,
+            CancellationToken cancellationToken = default)
+        {
+            return response.WriteAsJsonAsync<TValue>(value, options: null, contentType: JsonConstants.JsonContentTypeWithCharset, cancellationToken);
+        }
+
+        public static Task WriteAsJsonAsync<TValue>(
+            this HttpResponse response,
+            [AllowNull] TValue value,
+            JsonSerializerOptions? options,
+            CancellationToken cancellationToken = default)
+        {
+            return response.WriteAsJsonAsync<TValue>(value, options, contentType: JsonConstants.JsonContentTypeWithCharset, cancellationToken);
+        }
+
+        public static Task WriteAsJsonAsync<TValue>(
+            this HttpResponse response,
+            [AllowNull] TValue value,
+            JsonSerializerOptions? options,
+            string? contentType,
+            CancellationToken cancellationToken = default)
+        {
+            if (response == null)
+            {
+                throw new ArgumentNullException(nameof(response));
+            }
+
+            options ??= ResolveSerializerOptions(response.HttpContext);
+
+            if (contentType != null)
+            {
+                response.ContentType = contentType;
+            }
```

Comment:
Per earlier discussion with @Tratcher  - We should set `application/json; charset=utf-8` if content type wasn't provided.

Reply:
It seems weird to allow this to be null and do nothing if it's not passed.